subprojects/workers/src/main/java/org/gradle/workers/IsolationMode.java

```diff
@@ -19,22 +19,26 @@
 import org.gradle.api.Incubating;
 
 /**
- * Forking mode for workers.
+ * Isolation mode for workers.
  *
  * @since 3.5
  */
 @Incubating
-public enum ForkMode {
+public enum IsolationMode {
     /**
      * Let Gradle decide, this is the default.
      */
     AUTO,
     /**
-     * Never fork, aka. use in-process workers.
+     * Don't attempt to isolate the work, use in-process workers.
      */
-    NEVER,
+    NONE,
     /**
-     * Always fork, aka. use out-of-process workers.
+     * Isolate the work in it's own classloader, use in-process workers.
      */
-    ALWAYS;
+    CLASSLOADER,
+    /**
+     * Isolate the work in a separate process, use out-of-process workers.
+     */
+    PROCESS
```

Comment:
I'm pretty sure we won't want to tie up reusability into this boolean as there could be multiple combinations when we start to consider potentially having remote processes in the future - reusability would probably end up being a separate enum.  The potential remote case is probably a better comparison - I was thinking "PROCESS" vs "REMOTE", but maybe "EXTERNAL_PROCESS" vs "REMOTE_PROCESS" (future)?